assembly/src/ast/mod.rs

```diff
@@ -256,7 +287,14 @@ impl ModuleAst {
     /// Returns a new [ModuleAst].
     ///
     /// A module consists of internal and exported procedures but does not contain a body.
-    pub fn new(local_procs: Vec<ProcedureAst>, docs: Option<String>) -> Result<Self, ParsingError> {
+    pub fn new(
+        imports: BTreeMap<String, LibraryPath>,
+        local_procs: Vec<ProcedureAst>,
+        docs: Option<String>,
+    ) -> Result<Self, ParsingError> {
```

Comment:
minor nit: let's change the order of parameters here to:
```Rust
pub fn new(
    local_procs: Vec<ProcedureAst>,
    imports: BTreeMap<String, LibraryPath>,
    docs: Option<String>,
) -> Result<Self, ParsingError> {
```
And also update the order of fields in the struct accordingly.